node/smeshing_service_test.go

```diff
@@ -0,0 +1,240 @@
+			Module: "poet",
+			Level:  "DEBUG",
+		})
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+	})
+}
+
+func TestSmeshingService_StartSmeshing(t *testing.T) {
+	t.Run("starts smeshing with valid coinbase", func(t *testing.T) {
+		cfg := getSmeshingServiceTestConfig(t)
+		service, err := NewSmeshingService(cfg, zaptest.NewLogger(t))
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		err = service.startServices(ctx)
+		require.NoError(t, err)
+	})
+
+	t.Run("fails with invalid coinbase", func(t *testing.T) {
+		cfg := getSmeshingServiceTestConfig(t)
+		cfg.SMESHING.CoinbaseAccount = "invalid-address"
+
+		service, err := NewSmeshingService(cfg, zaptest.NewLogger(t))
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		err = service.startServices(ctx)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "parse CoinbaseAccount address")
+	})
+}
+
+func TestSmeshingService_GRPCServices(t *testing.T) {
+	t.Run("initializes all grpc services", func(t *testing.T) {
+		cfg := getSmeshingServiceTestConfig(t)
+		service, err := NewSmeshingService(cfg, zaptest.NewLogger(t))
+		require.NoError(t, err)
+
+		services := []grpcserver.Service{
+			grpcserver.Debug,
+			grpcserver.Smesher,
+			grpcserver.Post,
+			grpcserver.PostInfo,
+		}
+
+		for _, svc := range services {
+			api, err := service.grpcService(svc, zaptest.NewLogger(t))
+			require.NoError(t, err)
+			require.NotNil(t, api)
+		}
+	})
+
+	t.Run("fails with unknown service", func(t *testing.T) {
+		cfg := getSmeshingServiceTestConfig(t)
+		service, err := NewSmeshingService(cfg, zaptest.NewLogger(t))
+		require.NoError(t, err)
+
+		api, err := service.grpcService("unknown", zaptest.NewLogger(t))
+		require.Error(t, err)
+		require.Nil(t, api)
+	})
+}
+
+func TestSmeshingService_StopServices(t *testing.T) {
+	t.Run("stops all services", func(t *testing.T) {
+		cfg := getSmeshingServiceTestConfig(t)
+		service, err := NewSmeshingService(cfg, zaptest.NewLogger(t))
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		err = service.start(ctx)
+		require.NoError(t, err)
+
+		stopCtx, stopCancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer stopCancel()
+
+		service.stopServices(stopCtx)
+	})
+}
+
+func TestSmeshingService_PprofServer(t *testing.T) {
+	cfg := getSmeshingServiceTestConfig(t)
+	cfg.PprofHTTPServer = true
+	cfg.PprofHTTPServerListener = ":0"
+	cfg.PprofBlockProfile = true
+	cfg.PprofMutexProfile = true
+	service, err := NewSmeshingService(cfg, zaptest.NewLogger(t))
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	require.NoError(t, service.start(ctx))
+	defer func() {
+		cancel()
+		service.stopServices(context.Background())
+	}()
```

Comment:
Same here: why is both required? It should in my opinion only be either `cancel` or `stopServices`